### PR TITLE
Clean up unused parameters around Assembly::Load

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs [hunk 1]
```diff
@@ -670,9 +632,7 @@ public override Assembly GetSatelliteAssembly(CultureInfo culture, Version versi
             an.CultureInfo = culture;
             an.Name = GetSimpleName() + ".resources";
 
-            RuntimeAssembly retAssembly = nLoad(an, null, this, ref stackMark,
-                                IntPtr.Zero,
-                                throwOnFileNotFound);
+            RuntimeAssembly retAssembly = nLoad(an, null, ref stackMark, throwOnFileNotFound, IntPtr.Zero);
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: Why is it ok to not pass the requesting assembly here anymore?

Comment (marek-safar) [reply src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: I replaced it with https://github.com/dotnet/coreclr/pull/22716/files#diff-656cc6f286474bdd1af492899ad62123R621 instead

Comment (jkotas) [reply src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: `StackCrawlMark.LookForMe` in CoreLib will return CoreLib once resolved. So the requesting assembly is always going be CoreLib with this change. Different behavior.

Comment (marek-safar) [reply src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: I think that's same behaviour, before there was `.., this, ref stackMark, ..` where `stackMark` was unused and `this` was used. Now there is only 1 argument which is always this-like

Comment (filipnavara) [reply src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: I don't think it's the same behavior. You are trying to resolve a satellite assembly for some other assembly that comes from ResourceManager. For example, resolving to "<AppDirectory>\cs\Boo.resources.dll" when `this` is assembly representing `<AppDirectory>\Boo.dll`.

Comment (marek-safar) [reply src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: oh you are right, will update the PR

File: src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs [hunk 2]
```diff
@@ -613,12 +614,10 @@ public override Assembly GetSatelliteAssembly(CultureInfo culture, Version versi
             return InternalGetSatelliteAssembly(culture, version, true);
         }
 
-        [System.Security.DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod
```
Comment (filipnavara) [inline src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: This cannot be removed. It is enforced on checked builds for every method that has `StackCrawlMark` inside.

**Pull request merged**